src/Contacts/Contact.m

```diff
@@ -281,6 +281,22 @@ - (NSString *)nameForPhoneNumber:(NSString *)recipientId
     return value;
 }
 
+- (NSUInteger)hash
+{
+    NSUInteger hash = 1825038313 ^ self.fullName.hash;
+
+    // NSData.hash appears not to change even when the underlying bytes change.
+    // maybe it's built on address?
+    hash = hash ^ self.cnContact.thumbnailImageData.description.hash;
```

Comment:
> Contact has an emails property... do we use that anywhere? I think we do - shouldn't we include that in this hash so that changes to emails update the current contacts?

Good call.